department-of-veterans-affairs/caseflow
# Store cancelled by id on tasks

Bumps #14365

### Description
Adds cancelled by column to the task model adn sets the canceller when tasks are cancelled

### Acceptance Criteria
- [ ] The user id of the person who cancelled a task is stored on the task itself

### Testing Plan
1. Sign in as BVASCASPER
1. Go to your queue
1. Click on any case
1. Cancel your task
1. Go back to case details and look at the case timeline
1. Confirm your css id shows as the canceler.
1. Go to your queue
1. Click on any ama case
1. Remove all issues on the appeal
1. Go back to case details and look at the case timeline
1. Confirm your css id shows as the canceler on all cancelled tasks.
```ruby
Appeal.find_by(uuid: "ee6779fe-81b7-4a2b-95c2-e024b2439cfc").treee :status, :cancelled_by_id
                                ┌─────────────────────────────┐
Appeal 43 (direct_review) ───── │ STATUS    │ CANCELLED_BY_ID │
└── RootTask                    │ cancelled │ 1               │
    └── JudgeDecisionReviewTask │ cancelled │ 1               │
        └── AttorneyTask        │ cancelled │ 1               │
                                └─────────────────────────────┘
```

![Screen Shot 2020-09-03 at 3 13 31 PM](https://user-images.githubusercontent.com/45575454/92157369-527fc900-edf8-11ea-9b6b-833cd8ce2be0.png)


### Database Changes
*Only for Schema Changes*

* [ ] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
